libraries/Microsoft.Bot.Builder.Expressions/BuiltInFunctions.cs

```diff
@@ -1229,623 +1230,1733 @@ private static (object, string) ParseISOTimestamp(string timeStamp, Func<DateTim
+                                {
+                                    value = "morning";
+                                }
+                                else if (timestamp.Hour == 12 && timestamp.Minute == 0)
+                                {
+                                    value = "noon";
+                                }
+                                else if (timestamp.Hour < 18)
+                                {
+                                    value = "afternoon";
+                                }
+                                else if (timestamp.Hour < 22 || (timestamp.Hour == 22 && timestamp.Minute == 0))
+                                {
+                                    value = "evening";
+                                }
+                                else
+                                {
+                                    value = "night";
+                                }
                             }
                             return (value, error);
                         },
-                        VerifyInteger),
-                    ReturnType.Number,
-                    ValidateBinaryNumber),
+                        VerifyString),
+                    ReturnType.String,
+                    expr => ValidateOrder(expr, null, ReturnType.String)),
                 new ExpressionEvaluator(
-                    ExpressionType.Average,
-                    Apply(
-                        args =>
+                    ExpressionType.GetFutureTime,
+                    (expr, state) =>
+                    {
+                        object value = null;
+                        string error = null;
+                        IReadOnlyList<dynamic> args;
+                        (args, error) = EvaluateChildren(expr, state);
+                        if (error == null)
                         {
-                            List<object> operands = ResolveListValue(args[0]);
-                            return operands.Average(u => Convert.ToSingle(u));
-                        },
-                        VerifyNumericList),
-                    ReturnType.Number,
-                    ValidateUnary),
+                            if (args[0] is int int1 && args[1] is string string1)
+                            {
+                                var format = (args.Count() == 3) ? (string)args[2] : DefaultDateTimeFormat;
+                                Func<DateTime, DateTime> timeConverter;
+                                (timeConverter, error) = DateTimeConverter(int1, string1, false);
+                                if (error == null)
+                                {
+                                    value = timeConverter(DateTime.Now).ToString(format);
+                                }
+                            }
+                            else
+                            {
+                                error = $"{expr} can't evaluate.";
+                        
```

Comment:
args count condition can be merged into one :` var format = (args.Count() == 3) ? (string)args[2] : DefaultDateTimeFormat;`